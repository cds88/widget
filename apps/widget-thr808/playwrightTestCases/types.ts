export type E2ETestDescription<T> = {
  label: string;
  descriptionRequirement: string;
  testCaseKey: T;
};
