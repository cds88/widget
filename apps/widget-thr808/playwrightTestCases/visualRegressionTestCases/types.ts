export enum VisualRegressionE2ETestsEnum {
    PIXEL_MISMATCH_MAIN_VIEW = "PIXEL_MISMATCH_MAIN_VIEW"
}

