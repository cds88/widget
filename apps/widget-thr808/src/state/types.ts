export enum WidgetsEnum {
  WIDGET_THR08 = 'WIDGET_TH08',
}
