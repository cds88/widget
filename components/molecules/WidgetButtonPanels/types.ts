export type WidgetButtonPanelsOptionsLabels<T extends string> = Record< T, string>

 