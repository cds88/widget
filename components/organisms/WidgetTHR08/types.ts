export type WidgetTHR08State = {
    brightness:number;
    timeLeft: number;
    nightVision: boolean;
    duskTillDawn: boolean;
    flashing: boolean;

} 

export type WidgetTHR08ButtonPanelsOptionsKeys = keyof Pick<WidgetTHR08State, 'nightVision' | 'flashing' | 'duskTillDawn'>

