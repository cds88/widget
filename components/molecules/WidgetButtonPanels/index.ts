export { default } from './WidgetButtonPanels'
export type { WidgetButtonPanelsOptionsLabels } from './types'