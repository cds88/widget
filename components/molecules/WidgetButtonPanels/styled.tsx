import { Container, styled } from '@mui/material';

export const WidgetButtonPanelsContainer = styled(Container)`
  flex: 3;
  display: flex;
  justify-content: center;
  flex-direction: column;
`;
