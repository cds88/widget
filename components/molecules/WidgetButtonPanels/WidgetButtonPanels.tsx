import { ContainerProps } from '@mui/material';
import React from 'react';

 
import { WidgetButtonPanelsContainer } from './styled';
import { WidgetButtonPanelsOptionsLabels } from './types';
import { WIDGET_BUTTON_PANELS } from './const.test';   
import { WidgetButtonPanel } from '@widget/atoms';

 
 
 
 
 

type WidgetButtonPanelsProps<T extends string> = {
  onChange: (partialButtonPanelKeyValPair: Partial<Record<T, boolean>>) => void;
  widgetButtonPanelsOptionsLabels: WidgetButtonPanelsOptionsLabels<T>;
  state: Record<T, boolean>;
} & Omit<ContainerProps, 'onChange'>;

const WidgetButtonPanels = <T extends string>({
  onChange,
  widgetButtonPanelsOptionsLabels,
  state,
  ...props
}: WidgetButtonPanelsProps<T>) => {
  return (
    <WidgetButtonPanelsContainer {...props} data-testid={WIDGET_BUTTON_PANELS.WIDGET_BUTTON_PANELS_CONTAINER}>
      {Object.entries(widgetButtonPanelsOptionsLabels).map(([key, val]) => {
        const typedKey = key as T;
        const handleOnChange = (checked: boolean) => {
          onChange({
            [typedKey]: checked,
          } as Partial<Record<T, boolean>>);
        }
        return (
          <WidgetButtonPanel
            key={key}
            checked={state[typedKey]}
            onChange={handleOnChange}
            data-testid={`${WIDGET_BUTTON_PANELS.WIDGET_BUTTON_PANELS_PANEL_COMPONENT}${key}`}
          >
            {`${val}`}
          </WidgetButtonPanel>
        );
      })}
    </WidgetButtonPanelsContainer>
  );
};

export default WidgetButtonPanels;
