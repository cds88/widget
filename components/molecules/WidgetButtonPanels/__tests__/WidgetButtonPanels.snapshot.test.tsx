import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import WidgetButtonPanels from '../WidgetButtonPanels';
import { WidgetButtonPanelsOptionsLabels } from '../types';

type TestKeys = 'option1' | 'option2' | 'option3';
const widgetButtonPanelsOptionsLabels: WidgetButtonPanelsOptionsLabels<TestKeys> =
  {
    option1: 'Label 1',
    option2: 'Label 2',
    option3: 'Label 3',
  };

const state: Record<TestKeys, boolean> = {
  option1: true,
  option2: false,
  option3: true,
};

describe('WidgetButtonPanels', () => {
  const mockOnChange = jest.fn();
  const optionsLabels = {
    panel1: 'Panel 1',
    panel2: 'Panel 2',
  };

  beforeEach(() => {
    mockOnChange.mockClear();
  });

  test('should render the WidgetButtonPanels and match snapshot', () => {
    const { asFragment } = render(
      <WidgetButtonPanels
        onChange={mockOnChange}
        widgetButtonPanelsOptionsLabels={widgetButtonPanelsOptionsLabels}
        state={state}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('should render with all switches off', () => {
    const allOffState: Record<TestKeys, boolean> = {
      option1: false,
      option2: false,
      option3: false,
    };

    const { asFragment } = render(
      <WidgetButtonPanels
        onChange={mockOnChange}
        widgetButtonPanelsOptionsLabels={widgetButtonPanelsOptionsLabels}
        state={allOffState}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
