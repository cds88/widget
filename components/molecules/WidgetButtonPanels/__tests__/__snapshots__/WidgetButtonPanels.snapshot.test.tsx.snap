// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetButtonPanels should render the WidgetButtonPanels and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-63q09u-MuiContainer-root"
    data-testid="WIDGET_BUTTON_PANELS_CONTAINER"
  >
    <div
      class="MuiBox-root css-9c42oi"
      data-testid="WIDGET_BUTTON_PANELS_PANEL_COMPONENToption1"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
      >
        Label 1
      </h4>
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
    </div>
    <div
      class="MuiBox-root css-9c42oi"
      data-testid="WIDGET_BUTTON_PANELS_PANEL_COMPONENToption2"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
      >
        Label 2
      </h4>
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
    </div>
    <div
      class="MuiBox-root css-9c42oi"
      data-testid="WIDGET_BUTTON_PANELS_PANEL_COMPONENToption3"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
      >
        Label 3
      </h4>
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WidgetButtonPanels should render with all switches off 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-63q09u-MuiContainer-root"
    data-testid="WIDGET_BUTTON_PANELS_CONTAINER"
  >
    <div
      class="MuiBox-root css-9c42oi"
      data-testid="WIDGET_BUTTON_PANELS_PANEL_COMPONENToption1"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
      >
        Label 1
      </h4>
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
    </div>
    <div
      class="MuiBox-root css-9c42oi"
      data-testid="WIDGET_BUTTON_PANELS_PANEL_COMPONENToption2"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
      >
        Label 2
      </h4>
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
    </div>
    <div
      class="MuiBox-root css-9c42oi"
      data-testid="WIDGET_BUTTON_PANELS_PANEL_COMPONENToption3"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
      >
        Label 3
      </h4>
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
